app/controllers/observations/downloads_controller.rb

```diff
@@ -31,6 +33,15 @@ def print_labels
 
     private
 
+    def too_many_results
+      flash_error(:download_observations_too_many_results.t)
+      redirect_to(observations_path)
+    end
+
+    def too_many_results?
+      @query.flavor == :all
```

Comment:
Should I expand this? Examples:
- A fixed upper limit
- Max of a fixed upper limit || all the current user's Observations.